t/base/lex.t

```diff
@@ -25,14 +28,14 @@ $x = '\\'; # ';
 
 if (length($x) == 1) {print "ok 4\n";} else {print "not ok 4\n";}
 
-eval 'while (0) {
+eval 'no warnings q|uninitialized|; while (0) {
```

Comment:
> I'd probably move this out of the quote eval.

Why?  If you look at the revisions we did to get strict-by-default working, you'll see that we (or, at least, I) generally tried to relax strictures in the narrowest possible scope that was feasible.  This frequently meant inserting, say,  `no strict 'refs'` at the beginning of a string to be `eval`-ed.

Similarly, inserting the relaxation of warnings at the start of the string to be `eval`-ed is the narrowest possible scope.  (In this case, since the `eval` uses single-quotes, I had to change the single quotes inside to `q||`.)